Source/OWASP-2013-Demo.Domain/UserProvider.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Web;
+using System.Web.Script.Serialization;
+using System.Web.Security;
+using OWASP_2013_Demo.Interfaces.Entities;
+using OWASP_2013_Demo.Interfaces.Providers;
+using OWASP_2013_Demo.Interfaces.Repositories;
+using OWASP_2013_Demo.Interfaces.Utilities;
+using OWASP_2013_Demo.Models;
+
+namespace OWASP_2013_Demo.Domain
+{
+	public class UserProvider : IUserProvider
+	{
+		private readonly IUserRepository _customerRepository;
+		private readonly ISiteConfiguration _siteConfiguration;
+		private readonly IPasswordManager _passwordManager;
+
+		public string NoUserExistsError
+		{
+			get { return "Error: No user exists with that email address."; }
+		}
+		public string UserPasswordIncorrectError {
+			get { return "Error: Password is incorrect. Please try again."; }
+		}
+		public string UsernameOrPasswordIncorrectError {
+			get { return "Error: Username or password is incorrect. Please try again."; }
+		}
+
+		public IUserPrincipal User
```

Comment:
@SonOfSofaman This isn't working due to casting issues. Are you familiar with SecurityPrinical class? I'm going to go ahead and merge as this code isn't used yet but I'd like to get it working.